code-challenges.test.js

```diff
@@ -14,39 +16,238 @@
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
-// Expected output: true
+// const hand1 = [5, 5, 5, 3, 3]
+// const hand2 = [5, 5, 3, 3, 4]
+// const hand3 = [5, 5, 5, 5, 4]
+// const hand4 = [7, 2, 7, 2, 7]
+
+// describe("fullHouse", () => {
+//     it("creates a function that takes an array of 5 numbers and determines if the array is a full house; one pair and three of a kind", () => {
+//     expect(fullHouse(hand1)).toEqual("true")
+//     expect(fullHouse(hand2)).toEqual("false")
+//     expect(fullHouse(hand3)).toEqual("false")
+//     expect(fullHouse(hand4)).toEqual("true")
+// })
+// })
+
+
+//RED-Good Failure
+// FAIL  ./code-challenges.test.js
+// fullHouse
+//   ✕ creates a function that takes an array of 5 numbers and determines if the array is a full house; one pair and three of a kind (1 ms)
+
+// ● fullHouse › creates a function that takes an array of 5 numbers and determines if the array is a full house; one pair and three of a kind
+
+//   ReferenceError: fullHouse is not defined
+
+//     182 | describe("fullHouse", () => {
+//     183 |     it("creates a function that takes an array of 5 numbers and determines if the array is a full house; one pair and three of a kind", () => {
+//   > 184 |     expect(fullHouse(hand1)).toEqual("true")
+//         |     ^
+//     185 |     expect(fullHouse(hand2)).toEqual("false")
+//     186 |     expect(fullHouse(hand3)).toEqual("false")
+//     187 |     expect(fullHouse(hand4)).toEqual("true")
+
+//     at Object.expect (code-challenges.test.js:184:5)
+
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 total
+// Snapshots:   0 total
+// Time:        0.141 s, estimated 1 s
+// Ran all test suites.
 
 // b) Create the function that makes the test pass.
+//PseudoCode
+//Input:
+// const hand1 = [5, 5, 5, 3, 3]
+// const hand2 = [5, 5, 3, 3, 4]
+// const hand3 = [5, 5, 5, 5, 4]
+// const hand4 = [7, 2, 7, 2, 7]
+//Output: true, false, false, true
+//Process:
+//Create a function called fullHouse that passes an array as a parameter
+//create a for loop to iterate through the array
+//create conditional if statement stating if index values are the same *3 && the same *2, return true 
+//the internet resources said to use slice  before the array, inorder to take subset of the array?
+//.splice was another internet suggestion
+
+//my attempt
+// const fullHouse = (array) => {
+//     const newSubset = array.slice()
+//     for (let i=0; i <array.length; i++)
+//     if (array === ([i]))
+// }
+// console.log(fullHouse(hand1))
+
+
+//found on the internet for resource purposes. Don't really understand what's going on here though, especially at the splice level and the 'el'
```

Comment:
For now, I would not look into this, as it will only confuse you further. I suggest a different approach with different logic. For example, we can use .sort to sort the array from least to greatest. Let's take the hand1 variable for this example. We use .sort on hand1, it now becomes [3, 3, 5, 5, 5]. From there, we can use a conditional saying that if index 0 === index 1 and index 2 === index 3 && index 2 === index 4, that means we have a full house, because that means the first 2 numbers are the same, which is a pair, and the last 3 numbers are the same, which is a three of a kind. Then you would modify it to also work for something like [3, 3, 3, 5, 5].